src/ripple/proto/org/xrpl/rpc/v1/get_ledger.proto

```diff
@@ -58,6 +62,15 @@ message GetLedgerResponse
 
     // True if request was exempt from resource controls
     bool is_unlimited = 7;
+
+    // True if the response contains the state map diff
+    bool objects_included = 8;
+
+    // True if the response contains key of objects adjacent to objects in state
+    // map diff
+    bool object_neighbors_included = 9;
+
+    repeated BookSuccessor book_successors = 10;
```

Comment:
For `book_successors`, are you getting individual `Offer` objects successor and predecessors, or `DirectoryNode` predecessors and successors?

If it's the former, there's really no need to have that information in Clio. You can simply traverse `DirectoryNodes` similar to `ledger_data`, and then batch fetch all the offers, instead of traversing the offers like a linked list.